code/modules/research/machinery/_production.dm

```diff
@@ -206,6 +207,7 @@
  */
 /obj/machinery/rnd/production/proc/build_efficiency(path)
 	PRIVATE_PROC(TRUE)
+	SHOULD_BE_PURE(TRUE)
```

Comment:
What is the point of both this, and having the proc be private instead of protected.

Reply:
It probably doesn't need to be private, but all this is doing is preventing people from putting side efects in it

Reply:
None of these are actually nessassary but I use them now out of habit.

Private because I don't want subtypes to use this proc nor do I want it to be used outside by other objects as well(Currently this proc is used only inside this file so I thought why not make it Private, that's all)

Again just for decoration so why not